src/Majima/Parser.java

```diff
@@ -0,0 +1,96 @@
+package Majima;
+
+
+import Majima.command.AddTodoCommand;
+import Majima.command.Command;
+import Majima.command.ExitCommand;
+import Majima.command.ListCommand;
+import Majima.command.AddDeadlineCommand;
+import Majima.command.AddEventCommand;
+import Majima.command.MarkCommand;
+import Majima.command.DeleteCommand;
+import Majima.command.UnmarkCommand;
+import Majima.task.Task;
+
+public class Parser {
+
+    public static Command parse(String input) throws MajimaException {
+        String[] parts = input.split(" ", 2);
+        String command = parts[0];
+        String args = (parts.length > 1) ? parts[1] : "";
+
+        switch (command) {
+            case "list":
+                return new ListCommand();
+            case "todo":
+                return parseAddCommand(Task.TaskType.TODO, args);
+            case "deadline":
+                return parseAddCommand(Task.TaskType.DEADLINE, args);
+            case "event":
+                return parseAddCommand(Task.TaskType.EVENT, args);
+            case "delete":
+                return parseDeleteCommand(args);
+            case "mark":
+                return parseMarkCommand(args);
+            case "unmark":
+                return parseUnmarkCommand(args);
+            case "exit":
+                return new ExitCommand();
+            default:
+                throw new MajimaException("Kiryu? I ain't got the faintest idea of what ya just said!");
+        }
+    }
+
+    private static Command parseAddCommand(Task.TaskType type, String args) throws MajimaException {
+        String description;
+        switch (type) {
+            case TODO:
```

Comment:
Shouldn't have indentation for all cases

Reply:
Fixed, thanks